client/gutenberg/extensions/simple-payments/editor.js

```diff
@@ -27,6 +27,27 @@ registerBlockType( 'a8c/simple-payments', {
 		paymentId: { type: 'number' },
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'shortcode',
+				tag: 'simple-payment',
+				attributes: {
+					paymentId: {
+						type: 'number',
+						shortcode: ( { named: { id } } ) => {
+							if ( ! id ) {
+								return;
+							}
+
+							return parseInt( id, 10 );
```

Comment:
This could easily be `NaN`, in which case the attribute appears to wind up as `null`. Try pasting `[simple-payment id="foo"]`.

```suggestion
							const result === parseInt( id, 10 );
							if ( result ) {
								return result;
							}
```

We avoid `paymentId: null` in this way.